clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/UpsertAmiTagsAtomicOperation.groovy

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.netflix.spinnaker.clouddriver.aws.deploy.ops
+
+import com.amazonaws.services.ec2.model.CreateTagsRequest
+import com.amazonaws.services.ec2.model.DescribeImagesRequest
+import com.amazonaws.services.ec2.model.Filter
+import com.amazonaws.services.ec2.model.Tag
+import com.netflix.spinnaker.clouddriver.aws.deploy.description.UpsertAmiTagsDescription
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider
+import com.netflix.spinnaker.clouddriver.data.task.Task
+import com.netflix.spinnaker.clouddriver.data.task.TaskRepository
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation
+import org.springframework.beans.factory.annotation.Autowired
+
+class UpsertAmiTagsAtomicOperation implements AtomicOperation<Void> {
+  private static final String BASE_PHASE = "UPSERT_AMI_TAGS"
+
+  private static Task getTask() {
+    TaskRepository.threadLocalTask.get()
+  }
+
+  @Autowired
+  AmazonClientProvider amazonClientProvider
+
+  private final UpsertAmiTagsDescription description
+
+  UpsertAmiTagsAtomicOperation(UpsertAmiTagsDescription description) {
+    this.description = description
+  }
+
+  @Override
+  Void operate(List priorOutputs) {
+    def descriptor = "${description.credentials.name}/${description.amiName}"
+    task.updateStatus BASE_PHASE, "Initializing Upsert ASG Tags operation for ${descriptor}..."
+
+    description.regions.each { String region ->
```

Comment:
My thought was to have orca lookup the image and send down all regions that it exists in (vs having clouddriver figure it out) ... suppose it could go both ways.